### PR TITLE
Sw change fees to not null

### PR DESCRIPTION
**Pull request by SWvheerden**
Changed fees so that they cannot be null. Empty fees are counted as 0.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1,29 +1,29 @@
 [[package]]
 name = "actix"
-version = "0.7.5"
+version = "0.7.8"
```
Comment (mikethetike) [inline Cargo.lock]: I think undo the Cargo.lock mass upgrades

**Comment by SWvheerden** [comment]
check fee incoming parameters as well

File: db/src/models/orders.rs [hunk 2]
```diff
@@ -568,11 +568,11 @@ impl Order {
                 quantity: 1,
                 parent_id: None,
             };
-            if let Some(event_fee_in_cents) = event.fee_in_cents {
-                new_event_fee.unit_price_in_cents = event_fee_in_cents;
+            if event.fee_in_cents > 0 {
```
Comment (mikethetike) [inline db/src/models/orders.rs]: You can remove this whole `if` statement and replace with 
```suggestion

 new_event_fee.unit_price_in_cents = event.fee_in_cents;
                new_event_fee.commit(conn)?;	                
```

File: db/src/models/organizations.rs [hunk 3]
```diff
@@ -90,8 +90,8 @@ pub struct OrganizationEditableAttributes {
     pub postal_code: Option<String>,
     #[serde(default, deserialize_with = "deserialize_unless_blank")]
     pub phone: Option<String>,
-    #[serde(default, deserialize_with = "double_option::deserialize")]
-    pub event_fee_in_cents: Option<Option<i64>>,
+    #[serde(default, deserialize_with = "deserialize_unless_blank")]
```
Comment (mikethetike) [inline db/src/models/organizations.rs]: I think this attribute is only used for strings

File: db/src/models/ticket_instances.rs [hunk 4]
```diff
@@ -57,6 +57,7 @@ impl TicketInstance {
             .inner_join(wallets::table.on(ticket_instances::wallet_id.eq(wallets::id)))
             .inner_join(events::table.on(ticket_types::event_id.eq(events::id)))
             .filter(ticket_instances::id.eq(id))
+            .filter(events::status.ne(EventStatus::Draft.to_string()))
```
Comment (mikethetike) [inline db/src/models/ticket_instances.rs]: This has slipped in from a different PR

File: db/src/models/ticket_instances.rs [hunk 5]
```diff
@@ -99,6 +100,7 @@ impl TicketInstance {
             .inner_join(events::table.on(ticket_types::event_id.eq(events::id)))
             .filter(ticket_instances::id.eq(id))
             .filter(events::id.eq(event_id))
+            .filter(events::status.ne(EventStatus::Draft.to_string()))
```
Comment (mikethetike) [inline db/src/models/ticket_instances.rs]: Same

File: db/src/models/ticket_instances.rs [hunk 6]
```diff
@@ -140,6 +142,7 @@ impl TicketInstance {
                     end_time.unwrap_or_else(|| NaiveDate::from_ymd(3970, 1, 1).and_hms(0, 0, 0)),
                 ))
                 .filter(wallets::user_id.eq(user_id))
+                .filter(events::status.ne(EventStatus::Draft.to_string()))
```
Comment (mikethetike) [inline db/src/models/ticket_instances.rs]: Same

File: db/src/test/builders/event_builder.rs [hunk 7]
```diff
@@ -112,7 +112,7 @@ impl<'a> EventBuilder<'a> {
         .unwrap();
 
         let update_event_fee = EventEditableAttributes {
-            fee_in_cents: self.fee_in_cents,
+            fee_in_cents: self.fee_in_cents.unwrap_or(0),
```
Comment (mikethetike) [inline db/src/test/builders/event_builder.rs]: Is this needed?

Comment (SWvheerden) [reply db/src/test/builders/event_builder.rs]: yes this is, self in this case is optional.

File: db/migrations/00000000000051_make_fees_not_null/down.sql [hunk 8]
```diff
@@ -0,0 +1,9 @@
+ALTER TABLE organizations
+    ALTER COLUMN event_fee_in_cents DROP NOT NULL;
```
Comment (mikethetike) [inline db/migrations/00000000000051_make_fees_not_null/down.sql]:
```suggestion
    ALTER COLUMN event_fee_in_cents SET NULL;
```

File: db/migrations/00000000000051_make_fees_not_null/down.sql [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+ALTER TABLE organizations
+    ALTER COLUMN event_fee_in_cents DROP DEFAULT;
+
+ALTER TABLE events
+    ALTER COLUMN fee_in_cents DROP NOT NULL;
```
Comment (mikethetike) [inline db/migrations/00000000000051_make_fees_not_null/down.sql]:
```suggestion
    ALTER COLUMN fee_in_cents SET NULL;
```

File: db/migrations/00000000000051_make_fees_not_null/down.sql (hunk 8 above)
Comment (SWvheerden) [reply db/migrations/00000000000051_make_fees_not_null/down.sql]: up to you how we do this, I am happy both ways, this was just the way postgress documentation mentioned it.

**Pull request merged**